### ISSUE TITLE
Questions about Image Infilling pretraining task

### ISSUE THREAD

**Issue by varadgunjal** [OFA-Sys/OFA#279]
Hi, thank you for the fantastic work! I have several questions about details of the VQGAN for Image Infilling task setup since I'm trying to set it up as a pretraining task for OFA using custom data -

1. I understand from (https://github.com/OFA-Sys/OFA/issues/236) that OFA uses "VQGAN OpenImages (f=8), 8192, GumbelQuantization" from [taming-transformers](https://github.com/CompVis/taming-transformers). I want to confirm which weights are used in the pretraining process to generate the 256 image codes in `image_examples.tsv` (under `pretrain_data_examples.zip`) - are they from the listed model under [taming-transformers](https://github.com/CompVis/taming-transformers) or are they from the checkpoint released here [Finetuned checkpoint for Text-to-Image Generation on COCO && CLIP checkpoint && VQGAN checkpoint](https://ofa-beijing.oss-cn-beijing.aliyuncs.com/checkpoints/image_gen_large_best.zip)? The hash of both the files appears to be the same, but I wanted to make sure that using either of these would be correct.

2. From the discussion in https://github.com/OFA-Sys/OFA/issues/258 I understand that the image codes are only 256-long in `image_examples.tsv` since they represent only the central 128x128 portion of the image. I am unclear on the exact procedure used to generate these : are they generated by taking a center crop on the 256x256 input image and running that through the VQGAN Encoder or by generating the full 32x32 (1024) codes for the image and then simply picking those corresponding to the central patch of the image? Does the choice of procedure matter?

3. In the pretraining process, was there experimentation done with different types of masks viz. random masking, to compare against the central masking procedure?

4. Based on Table 11 in Appendix A, I understand that the pretraining task uses images from OpenImages, YFCC100M & ImageNet-21K. Am I correct in saying that for the Image Infilling task, the VQGAN model is initialized using the weights available under [taming-transformers](https://github.com/CompVis/taming-transformers) and then modified as the pretraining task progresses on these 3 datasets?

**Comment by jxst539246** [comment]
question 1 & 4: they are the same. we used the ckpt from  [taming-transformers](https://github.com/CompVis/taming-transformers) without fine-tuning.
question 2. We take a center crop on the 256*256 image to get a 128*128 patch and feed it to the VQGAN to generate 16*16 (256) codes.
question 3. We didn't use other masking strategies, due to the limitation of resources.

**Comment by varadgunjal** [comment]
Thanks for the info! Very helpful!

A follow-up question : Appendix A.1 states "We carefully filter our pretraining data and exclude images that appear in the validation and test sets of downstream tasks to avoid data leakage." Is it possible to share more details about this filtration process? Particularly for the Image Infilling task, it looks like the total size of the 3 datasets is >100M whereas the table lists OFA as only using ~36M images. So I was curious as to how so much of the data was left out.
